e2e/tests/stake_app_transfer.feature

```diff
@@ -2,21 +2,87 @@ Feature: App Stake Transfer Namespace
-        And this test ensures the "application" for account "app2" is not staked
-        And this test ensures the "application" for account "app3" is not staked
+        # Reduce the application unbonding period to avoid timeouts.
+        And an authz grant from the "gov" "module" account to the "pnf" "user" account for each module MsgUpdateParam message exists
+        And the "pnf" account sends an authz exec message to update the "shared" module param
+            | name                                  | value | type  |
+            | application_unbonding_period_sessions | 1     | int64 |
+        And an account exists for "app3"
         # Stake with 1 uPOKT more than the current stake used in genesis to make
         # the transaction succeed.
         And the account "app2" has a balance greater than "1000070" uPOKT
+        And the user successfully stakes a "application" with "1000070" uPOKT for "anvil" service from the account "app2"
+        # Begin transfer
+        When the user transfers the "application" stake from account "app2" to account "app3"
+        Then the user should be able to see standard output containing "txhash:"
+        And the user should be able to see standard output containing "code: 0"
+        And the pocketd binary should exit without error
+        And the user should wait for the "application" module "TransferApplication" message to be submitted
+        And the user should wait for the "application" module "TransferBegin" tx event to be broadcast
+        # The source application should still be staked and in the transfer period
+        And the "application" for account "app2" is staked with "1000070" uPOKT
+        And the application for account "app2" is in the "transfer" period
+        # The destination application is not created until the transfer period ends
+        And the user verifies the "application" for account "app3" is not staked
+        And the user should wait for the "application" module "TransferEnd" end block event to be broadcast
+        And the "application" for account "app3" is staked with "1000070" uPOKT
+        And the account balance of "app3" should be "0" uPOKT "less" than before
+        # The source application should be unstaked with no account balance
+        # change after the transfer period
+        And the user verifies the "application" for account "app2" is not staked
+        And the account balance of "app2" should be "0" uPOKT "more" than before
+        # Cleanup for other tests
+        # TODO(#): Until a network state reset API is implemented, we SHOULD
+        # manually unstake the applications to mitigate failures in other features
+        # which may otherwise make false assumptions about the starting state.
+        And the user successfully unstakes a "application" from the account "app3"
+
+    Scenario: Only one Application transfer with a given destination address in the same session will succeed
```

Comment:
This is the reason why I think we should allow transferring to an existing application.

Currently, having multiple transfers to the same destination address is non deterministic (depends on how the application list is iterated over in `transfer_applications.go`)

And it's being more problematic as the outcome is not known until the end of the transfer period.

If we agree on allowing multiple transfers/transfers to existing apps, let's just leave TODO.